code/modules/religion/sects/shadow_sect.dm

```diff
@@ -60,6 +63,73 @@
 	icon_state = "shadow-obelisk"
 	anchored = FALSE
 	break_message = "<span class='warning'>The Obelisk crumbles before you!</span>"
+	var/heal_delay = 20
+	var/last_heal = 0
+	var/spread_delay = 45
+	var/last_spread = 0
+
+/obj/structure/destructible/religion/shadow_obelisk/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	return ..()
+
+/obj/structure/destructible/religion/shadow_obelisk/process(delta_time)
+	var/datum/religion_sect/shadow_sect/sect = GLOB.religious_sect
+	if(!sect.faithful_used)
+		return
+	if(last_heal <= world.time)
+		last_heal = world.time + heal_delay
+		for(var/mob/living/L in range(5, src))
+			if(L.health == L.maxHealth)
+				continue
+			if(!isshadow(L) && !L.mind?.holy_role && !isfaithless(L))
+				continue
+			new /obj/effect/temp_visual/heal(get_turf(src), "#29005f")
+			if(isshadow(L) || L.mind?.holy_role || isfaithless(L))
+				L.adjustBruteLoss(-2*delta_time, 0)
+				L.adjustToxLoss(-2*delta_time, 0)
+				L.adjustOxyLoss(-2*delta_time, 0)
+				L.adjustFireLoss(-2*delta_time, 0)
+				L.adjustCloneLoss(-2*delta_time, 0)
+				L.updatehealth()
+				if(L.blood_volume < BLOOD_VOLUME_NORMAL)
+					L.blood_volume += 1.0
+			CHECK_TICK
```

Comment:
You probably don't need to CHECK_TICK here since this list will be relatively cheap. CHECK_TICK can come with its own problems since the tick allocation happens by byond instead of by the MC.

Reply:
Coolio, I'll remove it, I'll also pull it from the nature pylon too as it's almost the same code